### PR TITLE
fix: Update the Wallets page to put bank accounts at top.

### PR DESCRIPTION
**Pull request by Krishna2323**
### Fixed Issues
$ https://github.com/Expensify/App/issues/46242
PROPOSAL: https://github.com/Expensify/App/issues/46242#issuecomment-2251546887

### Tests
1. Go to `/settings/wallet`
2. Verify the design with the mock up attached below

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Go to `/settings/wallet`
2. Verify the design with the mock up attached below

### QA Steps
1. Go to `/settings/wallet`
2. Verify the design with the mock up attached below

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/b16501eb-dc4f-40cc-b1b4-5b6640c5fe93

https://github.com/user-attachments/assets/18c58275-e17c-4f55-8da6-91ad26079554

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/461f273a-f790-4e3e-b4fe-b8d72d780899

https://github.com/user-attachments/assets/ea4a0a70-5958-4770-8215-2953714c1982

Monosnap (9) New Expensify 2024-08-04 13-45-27

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/b04111bc-7a31-4106-8d99-46c7812fb695

https://github.com/user-attachments/assets/a6ebcd66-acc9-4dc3-b9c5-4338d818fb2a

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/2628c33b-4c97-4791-850c-2cffe0218ea8

https://github.com/user-attachments/assets/c2c000b3-2766-4c34-8d87-66452ecc76df

Monosnap (9) New Expensify 2024-08-04 13-45-27

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

web_chrome_wallet_enabled
web_chrome

Monosnap (9) New Expensify 2024-08-04 13-45-11

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/e0d6d222-7f31-4bbb-adf5-ceee54d94462

https://github.com/user-attachments/assets/d23b9031-1f75-423e-914d-ece291817bef

</details>

Monosnap  $250  Update the Wallets page to put bank accounts at top · Issue #46242 · Expensify:App 2024-07-28 21-28-47

### CODE REVIEW

**Comment by dubielzyk-expensify** [comment]
Could we get some mobile screenshots as well? 😄

**Comment by shawnborton** [comment]
Let us know when this is ready for review. Also, do you have the correct design asset for this? I think the screenshot in the original comment is actually just a mockup from Figma, right?

**Comment by shawnborton** [comment]
Sounds good, here is the asset you need which should be shown at 262x152: [emptystate__big-vault.svg.zip](https://github.com/user-attachments/files/16408752/emptystate__big-vault.svg.zip)

**Comment by dubielzyk-expensify** [comment]
According to the mock in [this issue](https://github.com/Expensify/App/issues/46242) the currency should be using our `Expensify New Kansas` typeface. I'm not 100% sure the exact style/class to apply but it's the one we use on a report/expense screen.

CleanShot 2024-08-01 at 10 23 00@2x

Here's the mock from @shawnborton :
CleanShot 2024-08-01 at 10 23 33@2x

**Comment by shawnborton** [comment]
Yup, I'm using our standard h1 styles in Figma for that one. But my thinking is to make it look a bit more like the amount field in the expense view elsewhere in the app: 
CleanShot 2024-08-01 at 06 11 16@2x

...which uses h2, so maybe we just use h2 here and we get this:
image

**Comment by shawnborton** [comment]
@trjExpensify @anmurali @danielrvidal @kevinksullivan did we decide that we want the assigned cards section to be at the very bottom, or right below bank accounts?

**Comment by shawnborton** [comment]
I'm not entirely sure how contributors can enable a wallet. @joekaufmanexpensify do you have any ideas there?

From my testing though, it looks like the transfer balance page is working as expected:

https://github.com/user-attachments/assets/e3a61785-f49b-4c71-9cc9-fce5c0b37726

**Comment by shawnborton** [comment]
I think h2 so it's consistent with how the expense view displays totals.

**Review by paultsimura — CHANGES_REQUESTED** [review]

**Comment by joekaufmanexpensify** [comment]
@shawnborton I don't think there's a way for them to test wallets now unless they have a US address and SSN. We are working on creating a set of test wallet credentials that can be used [here](https://github.com/Expensify/Expensify/issues/364990).

**Comment by shawnborton** [comment]
I would say for titles, we do not want the trailing period.

**Review by paultsimura — APPROVED** [review]

**Comment by shawnborton** [comment]
I will bump them internally.

**Comment by neil-marcellini** [comment]
I will review once the latest batch of requested changes are addressed. I'll check back on Monday.

**Comment by shawnborton** [comment]
Where can I find updated screenshots that also included an assigned cards section?

**Comment by shawnborton** [comment]
Okay, let me spin up a test build and I can test it on my account.

**Review by neil-marcellini — APPROVED** [review]
The code looks fine to me so good to go pending approval from design. @shawnborton @dubielzyk-expensify

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
Looks good from a design perspective. I think this can be merged!

File: src/pages/settings/Wallet/WalletPage/WalletPage.tsx
Comment (parasharrajat) [inline src/pages/settings/Wallet/WalletPage/WalletPage.tsx:389]: The wrong vertical margin caused https://github.com/Expensify/App/issues/64926

**Pull request merged**
